sbkeys/README.md

```diff
@@ -0,0 +1,134 @@
+
+```shell
+ARCH="$(uname -m)"
+SDK_VERSION="v0.29.0"
+./generate-local-sbkeys \
+  --sdk-image "public.ecr.aws/bottlerocket/bottlerocket-sdk-${ARCH}:${SDK_VERSION}" \
+  --output-dir "${PWD}/my-local-profile"
+```
+
+## Create a profile with AWS-based resources
+
+The `generate-aws-sbkeys` tool can be used to create an AWS-based Secure Boot Keys profile.
+
+It uses the AWS CLI and [aws-kms-pkcs11](https://github.com/JackOfMostTrades/aws-kms-pkcs11) to obtain certificates, and the `virt-fw-vars` tool to generate EFI variable data.
+It creates an `aws-kms-pkcs11` configuration file for subsequent signing operations.
+
+When specifying an SDK image, these dependencies are run within a container started from that image, and do not need to be installed on the host.
+
+The tool expects four [AWS Private CAs](https://docs.aws.amazon.com/privateca/latest/userguide/PcaWelcome.html) and three [AWS KMS asymmetric keys](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#asymmetric-keys-concept) to be available.
+Note that the cost of these resources, **especially the private CAs**, is nontrivial.
+
+Although it is possible to use the same private CA and the same KMS key for all roles, doing so would weaken the security of the implementation, and is **strongly discouraged**.
+
+```shell
+ARCH="$(uname -m)"
+SDK_VERSION="v0.29.0"
+
+# AWS Private CAs
+PK_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/11111111-1111-1111-1111-111111111111"
+KEK_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/22222222-2222-2222-2222-222222222222"
+DB_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/33333333-3333-3333-3333-333333333333"
+VENDOR_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/44444444-4444-4444-4444-444444444444"
+
+# AWS KMS asymmetric keys
+SHIM_SIGN_KEY="arn:aws:kms:us-west-2:999999999999:key/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
+CODE_SIGN_KEY="arn:aws:kms:us-west-2:999999999999:key/bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb"
+CONFIG_SIGN_KEY="arn:aws:kms:us-west-2:999999999999:key/cccccccc-cccc-cccc-cccc-cccccccccccc"
+
+./generate-aws-sbkeys \
+  --sdk-image "public.ecr.aws/bottlerocket/bottlerocket-sdk-${ARCH}:${SDK_VERSION}" \
+  --aws-region us-west-2 \
+  --pk-ca "${PK_CA}" \
+  --kek-ca "${KEK_CA}" \
+  --db-ca "${DB_CA}" \
+  --vendor-ca "${VENDOR_CA}" \
+  --shim-sign-key "${SHIM_SIGN_KEY}" \
+  --code-sign-key "${CODE_SIGN_KEY}" \
+  --config-sign-key "${CONFIG_SIGN_KEY}" \
+  --output-dir "${PWD}/my-aws-profile"
+```
+
+To generate the profile, the IAM user or role should have a policy like this:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "acm-pca:GetCertificate",
+        "acm-pca:GetCertificateAuthorityCertificate",
+        "acm-pca:IssueCertificate",
+        "kms:GetPublicKey",
+        "kms:Sign"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
```

Comment:
nit: probably in the future we could narrow down the permissions in the policy to be the CAs used for the build. Same in other policies.

Reply:
> nit: probably in the future we could narrow down the permissions in the policy to be the CAs used for the build. Same in other policies.

I'd prefer to leave that as an exercise to the reader, if it's important to them, since they'll need to fill in the correct ARNs.